mlflow/pipelines/steps/train.py

```diff
@@ -317,7 +318,13 @@ def my_warn(*args, **kwargs):
                 )
                 if os.path.exists(output_model_path) and os.path.isdir(output_model_path):
                     shutil.rmtree(output_model_path)
-                mlflow.sklearn.save_model(model, output_model_path)
+
+                if self.output_probabilities:
+                    mlflow.sklearn.save_model(
+                        model, output_model_path, pyfunc_predict_fn="predict_proba"
+                    )
+                else:
+                    mlflow.sklearn.save_model(model, output_model_path)
```

Comment:
Looks like we're logging the model to MLflow Tracking on lines 311 - 313 and then saving the model to local storage with a different pyfunc predict function here. Should we be extending lines 311 - 313 instead of adding a separate `save_model` call here?

We should also make sure that the model schema is adjusted based on whether or not we're outputting probabilities.

Reply:
Thanks for calling this out. I've updated the call to `mlflow.sklearn.log_model` to use predict_proba and use the updated schema. I think that we need to also keep these changes in `mlflow.sklearn.save_model` if a user wants to load from local.